src/devices_models/devices/common/range_constraint.jl

```diff
@@ -152,6 +152,159 @@ function add_semicontinuous_range_constraints!(
+
+``` varcts[name, t] >= limits.min * (1 - varbin[name, t]) ```
+
+where limits in constraint_infos.
+
+# LaTeX
+
+`` 0 \leq x^{cts} \leq limits^{max} (1 - x^{bin}), \text{ for } limits^{min} = 0 ``
+
+`` limits^{min} (1 - x^{bin}) \leq x^{cts} \leq limits^{max} (1 - x^{bin}), \text{ otherwise } ``
+"""
+function add_reserve_range_constraints!(
+    container::OptimizationContainer,
+    T::Type{InputActivePowerVariableLimitsConstraint},
+    U::Type{<:VariableType},
+    devices::IS.FlattenIteratorWrapper{V},
+    model::DeviceModel{V, W},
+    X::Type{<:PM.AbstractPowerModel},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+) where {V <: PSY.Component, W <: AbstractDeviceFormulation}
+    use_parameters = built_for_simulation(container)
+    constraint = T()
+    variable = U()
+    component_type = V
+    time_steps = get_time_steps(container)
+    jump_variable = get_variable(container, variable, component_type)
+    names = [PSY.get_name(d) for d in devices]
+    binary_variables = [ReservationVariable()]
+
+    @assert length(binary_variables) == 1
+    varbin = get_variable(container, only(binary_variables), component_type)
+
+    names = [PSY.get_name(x) for x in devices]
+    # MOI has a semicontinous set, but after some tests is not clear most MILP solvers support it.
+    # In the future this can be updated
+    con_ub = add_cons_container!(
+        container,
+        constraint,
+        component_type,
+        names,
+        time_steps,
+        meta = "ub",
+    )
+    con_lb = add_cons_container!(
+        container,
+        constraint,
+        component_type,
+        names,
+        time_steps,
+        meta = "lb",
+    )
+
+    for device in devices, t in time_steps
+        ci_name = PSY.get_name(device)
+        limits = get_min_max_limits(device, T, W)
+        if JuMP.has_lower_bound(jump_variable[ci_name, t])
+            JuMP.set_lower_bound(jump_variable[ci_name, t], 0.0)
+        end
+        expression_ub = JuMP.AffExpr(0.0, jump_variable[ci_name, t] => 1.0)
+        expression_lb = JuMP.AffExpr(0.0, jump_variable[ci_name, t] => 1.0)
+        con_ub[ci_name, t] = JuMP.@constraint(
+            container.JuMPmodel,
+            expression_ub <= limits.max * (1 - varbin[ci_name, t])
+        )
+        con_lb[ci_name, t] = JuMP.@constraint(
+            container.JuMPmodel,
+            expression_lb >= limits.min * (1 - varbin[ci_name, t])
+        )
+    end
+end
+
+@doc raw"""
+Constructs min/max range constraint from device variable and reservation decision variable.
+
+# Constraints
+
+``` varcts[name, t] <= limits.max * varbin[name, t] ```
+
+``` varcts[name, t] >= limits.min * varbin[name, t] ```
+
+where limits in constraint_infos.
+
+# LaTeX
+
+`` limits^{min} x^{bin} \leq x^{cts} \leq limits^{max} x^{bin},``
+"""
+function add_reserve_range_constraints!(
+    container::OptimizationContainer,
+    T::Type{<:PowerVariableLimitsConstraint},
```

Comment:
Should this be `OutputActivePowerVariableLimitsConstraint`

Reply:
This is also used for the ReactivePowerLimits constraints, that's why I left it like that